staging/src/k8s.io/apiserver/pkg/storageversion/manager.go

```diff
@@ -112,22 +124,28 @@ func (s *defaultManager) UpdateStorageVersions(kubeAPIServerClientConfig *rest.C
 	sc := clientset.InternalV1alpha1().StorageVersions()
 
 	s.mu.RLock()
-	resources := []*ResourceInfo{}
+	resources := []ResourceInfo{}
 	for resource := range s.managedResourceInfos {
-		resources = append(resources, resource)
+		resources = append(resources, *resource)
 	}
 	s.mu.RUnlock()
 	hasFailure := false
-	for _, r := range resources {
+	sortResourceInfosByGroupResource(resources)
```

Comment:
why do we sort?

Reply:
To make sure we have a consistent dedup result, and therefore avoid creating unnecessarily duplicated StorageVersion objects. 

For example, `extensions.ingresses` and `networking.k8s.io.ingresses` share the same underlying storage. Without sorting, in an HA cluster, one apiserver may dedup and update StorageVersion for `extensions.ingresses`, while another apiserver may dedup and update StorageVersion for `networking.k8s.io.ingresses`. The storage migrator (which migrates objects per GroupResource) will migrate these resources twice, since both StorageVersion objects have `CommonEncodingVersion` (each with one server registered).

Reply:
Added the explanation to comments.